Sources/PackageCollections/Storage/SQLitePackageCollectionsStorage.swift

```diff
@@ -544,23 +663,49 @@ final class SQLitePackageCollectionsStorage: PackageCollectionsStorage, Closable
         """
         try db.exec(query: table)
 
-        let ftsPackages = """
-            CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.packagesFTSName) USING fts4(
-                collection_id_blob_base64, id, version, name, repository_url, summary, keywords, products, targets,
-                notindexed=collection_id_blob_base64,
-                tokenize=unicode61
-            );
-        """
-        try db.exec(query: ftsPackages)
-
-        let ftsTargets = """
-            CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.targetsFTSName) USING fts4(
-                collection_id_blob_base64, package_repository_url, name,
-                notindexed=collection_id_blob_base64,
-                tokenize=unicode61
-            );
-        """
-        try db.exec(query: ftsTargets)
+        do {
+            let ftsPackages = """
+                CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.packagesFTSName) USING fts4(
+                    collection_id_blob_base64, id, version, name, repository_url, summary, keywords, products, targets,
+                    notindexed=collection_id_blob_base64,
+                    tokenize=unicode61
+                );
+            """
+            try db.exec(query: ftsPackages)
+
+            let ftsTargets = """
+                CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.targetsFTSName) USING fts4(
+                    collection_id_blob_base64, package_repository_url, name,
+                    notindexed=collection_id_blob_base64,
+                    tokenize=unicode61
+                );
+            """
+            try db.exec(query: ftsTargets)
+
+            useSearchIndices.put(true)
+        } catch {
+            // These DDL statements work but queries yield different results when run on different
+            // platforms. This could be because of SQLite version perhaps? But since we can't get
+            // consistent results we will not fallback to FTS3 and just give up if FTS4 is not available.
+            /*
+             let ftsPackages = """
+                 CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.packagesFTSName) USING fts3(
+                     collection_id_blob_base64, id, version, name, repository_url, summary, keywords, products, targets,
+                     tokenize=simple
+                 );
+             """
+             try db.exec(query: ftsPackages)
+
+             let ftsTargets = """
+                 CREATE VIRTUAL TABLE IF NOT EXISTS \(Self.targetsFTSName) USING fts3(
+                     collection_id_blob_base64, package_repository_url, name,
+                     tokenize=porter
+                 );
+             """
+             try db.exec(query: ftsTargets)
```

Comment:
should we remove this? or keep around so we can fix later?

Reply:
removed https://github.com/apple/swift-package-manager/pull/3213/commits/a84e786e1d2fab0254c3d21327a0e79eb8f4e642